example/logsignature_example.py

```diff
@@ -0,0 +1,102 @@
+######################
+# Here's how to code up a Neural CDE using the log-ode method for a long time series.
+# This method assumes familiarity with the standard Neural CDE example at `example.py`. We will only describe the
+# differences from that example.
+######################
+import time
+import torch
+import torchcde
+from time_series_classification import NeuralCDE, get_data
+
+torch.manual_seed(0)
+
+
+def _train(train_X, train_y, test_X, test_y, depth=None, num_epochs=10, window_length=100):
+    # Time the training process
+    start_time = time.time()
+
+    # Logsignature computation step
+    train_logsig = torchcde.logsig_windows(train_X, depth, window_length=window_length)
+    print("Logsignature shape: {}".format(train_logsig.size()))
+
+    model = NeuralCDE(
+        input_channels=train_logsig.size(-1), hidden_channels=8, output_channels=1, interpolation="linear"
+    )
+    optimizer = torch.optim.Adam(model.parameters(), lr=0.1)
+
+    train_coeffs = torchcde.linear_interpolation_coeffs(train_logsig)
+
+    train_dataset = torch.utils.data.TensorDataset(train_coeffs, train_y)
+    train_dataloader = torch.utils.data.DataLoader(train_dataset, batch_size=32)
+    for epoch in range(num_epochs):
+        for batch in train_dataloader:
+            batch_coeffs, batch_y = batch
+            pred_y = model(batch_coeffs).squeeze(-1)
+            loss = torch.nn.functional.binary_cross_entropy_with_logits(pred_y, batch_y)
+            loss.backward()
+            optimizer.step()
+            optimizer.zero_grad()
+        print("Epoch: {}   Training loss: {}".format(epoch, loss.item()))
+
+    # Remember to compute the logsignatures of the test data too!
+    test_logsig = torchcde.logsig_windows(test_X, depth, window_length=window_length)
+    test_coeffs = torchcde.linear_interpolation_coeffs(test_logsig)
+    pred_y = model(test_coeffs).squeeze(-1)
+    binary_prediction = (torch.sigmoid(pred_y) > 0.5).to(test_y.dtype)
+    prediction_matches = (binary_prediction == test_y).to(test_y.dtype)
+    proportion_correct = prediction_matches.sum() / test_y.size(0)
+    print("Test Accuracy: {}".format(proportion_correct))
+
+    # Total time
+    elapsed = time.time() - start_time
+
+    return proportion_correct, elapsed
+
+
+def print_heading(message):
+    # Print a message inbetween rows of #'s
+    string_sep = "#" * 50
+    print("\n" + string_sep + "\n{}\n".format(message) + string_sep)
+
+
+def main(num_epochs=15):
```

Comment:
Is this a sensible `num_epochs` to get good results? It feels quite small.

Reply:
depth 1 gets 78% accuracy, depth 2 gets 98% accuracy, at least for `torch.manual_seed(0)`.